devices/zyxel-nwa50ax/default.nix

```diff
@@ -0,0 +1,310 @@
+      gcc = {
+        abi = "32";
+        arch = "mips32";          # mips32r2?
+      };
+    };
+  };
+
+  description = ''
+    Zyxel NWA50AX
+    ********************
+
+    Zyxel NWA50AX is quite close to the GL-MT300N-v2 "Mango" device, but it is based on the MT7621
+    chipset instead of the MT7628.
+
+    Installation
+    ============
+
+    Wow, this device is a handful.
+
+    The stock vendor firmware is a downstream fork of U-Boot (insert link to source code)
+    with restricted boot commands. Fortunately, OpenWrt folks figured out trivial command injections,
+    so you can use most of the OpenWrt commands without trouble by just command injecting
+    atns, atna or atnf, e.g. atns "; $real_command".
+
+    From factory web UI, this has not been tested yet.
+    From serial, you have two choices:
+
+    - boot from an existing Liminix system, e.g. TFTPBOOT image.
+    - boot from an OpenWrt system, i.e. follow OpenWrt steps.
+
+    Once you are in a Linux system, understand that this device has A/B boot.
+
+    OpenWrt provides you with `zyxel-bootconfig` to set/unset the image status and choice.
+    The kernel is booted with `bootImage=<number>` which tells you which slot are you on.
+
+    You should find yourself with 10ish MTD partitions, the most interesting ones are two:
+
+    - firmware: 40MB
+    - firmware_1: 40MB
+
+    In the current setup, they are split further into kernel (8MB) and ubi (32MB).
+
+    You will write the FIT uImage in the kernel MTD partition directly, via `mtd write kernel.uimage /dev/mtd4`
+    (if `/dev/mtd4` is the kernel MTD partition) or `flashcp` can be used similarly.
+
+    Then, you will install the UBI image, via `ubiformat -f ubi.image /dev/mtd5` (if `/dev/mtd4` was your kernel MTD partition, 5 must be the UBI).
+
+    Once you are done, you will have to reboot and modify your bootargs in U-Boot
+    because they are currently overridden.
+
+    It is enough to do:
+    atns "; setenv bootargs 'liminix console=ttyS0,115200 panic=10 oops=panic init=/bin/init loglevel=8 fw_devlink=off ubi.mtd=ubi root=ubi0:rootfs rootfstype=ubifs'"
+    atns "; saveenv"
+
+    Once you are done, you can enjoy booting on slot A unattended for now. To make use of the A/B feature, this is a TODO :).
+    Upgrades are TODO too.
+
+    Vendor web page: https://www.zyxel.com/fr/fr/products/wireless/ax1800-wifi-6-dual-radio-nebulaflex-access-point-nwa50ax
+
+    OpenWrt web page: https://openwrt.org/inbox/toh/zyxel/nwa50ax
+    OpenWrt tech data: https://openwrt.org/toh/hwdata/zyxel/zyxel_nwa50ax
+
+  '';
+
+  module = { pkgs, config, lib, lim, ...}:
+    let
+      inherit (pkgs.liminix.networking) interface;
+      inherit (pkgs.liminix.services) oneshot;
+      inherit (pkgs.pseudofile) dir symlink;
+      inherit (pkgs) openwrt;
+
+      mac80211 = pkgs.mac80211.override {
+        drivers = [ "mt7915e" ];
+        klibBuild = config.system.outputs.kernel.modulesupport;
+      };
```

Comment:
Note: The Future here is to switch from  pkgs.mac80211 (which uses backports) to pkgs.kmodloader, which uses the modules in the current kernel.See  https://gti.telent.net/dan/liminix/commit/65479e206b62a9b269d3825f8cc5e9d660d583ec 

Caveat: I haven't yet tested it on a mt7915e and it may be that by doing this we've lost necessary openwrt patches. See  e.g. https://gti.telent.net/dan/liminix/commit/ca64e9035e0e9792059faa5fcaaee2b959bed458 for an example of restoring them